pkg/oidc/provider/provider.go

```diff
@@ -76,11 +80,41 @@ func NewProvider(ctx context.Context, tokenCache wrangmgmtv3.TokenCache, tokenCl
 	}, nil
 }
 
+func (p *Provider) addHeadersMiddleware(next http.HandlerFunc) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Set("X-Content-Type-Options", "nosniff")
+		w.Header().Set("X-Frame-Options", "SAMEORIGIN")
+		w.Header().Set("Referrer-Policy", "strict-origin-when-cross-origin")
+		w.Header().Set("Strict-Transport-Security", "max-age=31536000; includeSubDomains; preload")
```

Comment:
For this set of headers, why did you choose these values? For example, would we never want to configure `max-age` to be a shorter time?

Reply:
These values are suggested by the security team https://github.com/rancher/security-team/issues/1343#issuecomment-2808870456

Reply:
Upon further investigation into this, we won't be able to use `preload`, so `max-age` no longer needs to be at least `31536000`.

```suggestion
		w.Header().Set("X-Content-Type-Options", "nosniff")
		w.Header().Set("X-Frame-Options", "SAMEORIGIN")
		w.Header().Set("Referrer-Policy", "strict-origin-when-cross-origin")
		w.Header().Set("Strict-Transport-Security", "max-age=31536000; includeSubDomains")
```